routers/web/web.go

```diff
@@ -1584,6 +1585,18 @@ func registerRoutes(m *web.Router) {
 			m.Get("/*", context.RepoRefByType(""), repo.Home) // "/*" route is deprecated, and kept for backward compatibility
 		}, repo.SetEditorconfigIfExists)
 
+		// Add a /tree/* path to redirect to the deprecated /src/*
+		// path. This emulates both Github & Gitlab's URL structure.
+		m.Get("/tree/*", func(ctx *context.Context) {
+			prefix := "/"
+			if setting.AppSubURL != "" {
+				prefix = setting.AppSubURL
+			}
+
+			url := path.Join(prefix, ctx.PathParam("username"), ctx.PathParam("reponame"), "src", ctx.PathParam("*"))
```

Comment:
It needs escaping.

And please move this handler into a proper "go" file instead of bloating the "web.go".

Reply:
Fixed in https://github.com/go-gitea/gitea/pull/33424/commits/dcf59838f6842f6bb4980ea500398894e2596452